### PR TITLE
IA-4897: add authorization to admin endpoints

### PR DESCRIPTION
**Pull request by cpate4**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/IA-4897

### What

- add a request interceptor to ensure only authenticated users, _and_ 
- ensure only two service accounts have access to these admin endpoints (leonardo and appmanager)

### Why

- locking down endpoints for authorized use only

## Checklist

- [ ] Thorough tests have been added and/or updated for this change
- [ ] Documentation has been updated for this change 
- [ ] This change has been validated in a BEE and/or locally 
- [ ] Primary reviewer validated this change

### CODE REVIEW

**Comment by cpate4** [comment]
unit tests and documentation are being updated ... 
those changes are still TBD

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -86,6 +88,29 @@ With IntelliJ IDEA open, perform the following steps to set up your IDE:
...
+Following the details outlined in the [DESIGN.md](./DESIGN.md),
+this service assumes all requests are authenticated by the time they reach this service.
+To test this functionality locally, you need to make sure the Apache proxy is running locally.
+
+The `/admin`-endpoints are locked down to only allow specific service accounts to access them.
+
+To impersonate service account in development, one must:
+
+```shell
+gcloud config set auth/impersonate_service_account appmanager-dev@broad-dsde-dev.iam.gserviceaccount.com
```
Comment (jdcanas) [inline CONTRIBUTING.md:102]: do all of our broad accounts have permission to impersonate this service account by default?

Comment (cpate4) [reply CONTRIBUTING.md:102]: `sherlock` does, and we are authorizing `leonardo` and `appmanager` in our helm charts

Comment (jdcanas) [reply CONTRIBUTING.md:102]: I guess I'm wondering what gcloud auth specifically I have to do to get admin access and use this locally

Comment (cpate4) [reply CONTRIBUTING.md:102]: `gcloud auth` as your `broad institute.org` account, and then follow the steps in the `CONTRIBUTING.md` document. did that work for you?

Comment (jdcanas) [reply CONTRIBUTING.md:102]: Just making sure I understood the steps 👍

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 # This is for deployment-specific values, which may be managed by other teams
 
 env:
+  admin:
```
Comment (lucymcnatt) [inline service/src/main/resources/application.yml:5]: just to clarify, these service accounts are specified by through the helmfile?

**Review by lucymcnatt — APPROVED** [review]

File: service/src/main/resources/application.yml (hunk 2 above)
Comment (cpate4) [reply service/src/main/resources/application.yml:5]: yes ... we are doing that here:
https://github.com/broadinstitute/terra-helmfile/pull/5628

Comment (lucymcnatt) [reply service/src/main/resources/application.yml:5]: oh yes, sorry! I remembered seeing that but couldn't find it looking through helmfile commits (or course its still a PR)

Comment (cpate4) [reply service/src/main/resources/application.yml:5]: it's all good! there are so many PRs and code reviewed ... this seems like light-years ago 🚀

**Review by cindy-broadinstitute — APPROVED** [review]

File: service/src/main/java/bio/terra/appmanager/controller/AdminController.java [hunk 3]
```diff
@@ -48,8 +55,6 @@ public ResponseEntity<Void> deleteChart(String body) {
     return ResponseEntity.noContent().build();
   }
 
-  // Note that this method's implementation relies on `includeAll` having a default value and being
-  // not null
```
[the remark this answers is not in the document]
Comment (cpate4) [reply service/src/main/java/bio/terra/appmanager/controller/AdminController.java:52]: I'll add it back in. A default value is currently set in swagger.

**Pull request merged**